src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingVisitor.cs

```diff
@@ -141,6 +142,18 @@ public override void VisitMarkupBlock(MarkupBlockSyntax node)
 
         public override void VisitMarkupElement(MarkupElementSyntax node)
         {
+            // If this is the direct child of a @section directive then we add a Razor indent to essentially
+            // set the baseline one level in.
+            // Note: "Direct child" in a little poetic license, as the structure is actually:
+            //     MarkupElement -> MarkupBlock -> CSharpCodeBlock (¯\_(ツ)_/¯) -> RazorDirectiveBody -> RazorDirective
+            var inSectionDirective = (node.Parent?.Parent?.Parent?.Parent is RazorDirectiveSyntax directive &&
```

Comment:
Does `FirstAncestorOrSelf<RazorDirectiveSyntax>` work here?

Reply:
It would work here, but also walks the tree all the way to the top, for every node, so would be quite a bit less efficient. Given we know the structure, I think its better to enforce it directly. There is probably an argument to go even further and pattern match against every level of the tree (ie `if (node.Parent is MarkupBlock { Parent: CSharpCodeBlock { Parent: RazorDirectiveBody` etc. etc.

Reply:
Okay, can't do that, but I can do `MarkupBlock`